File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    
    const publication = [];
    
    for(let i = 1; i <= score.length; i++){
        // 매일 순위 목록
        const sort = score.slice(0, i).sort((a, b) => b - a);
        
        publication.push(i < k ? sort[i - 1] : sort[k-1])
    }
    
    return publication;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    hallOfFame = []
    for i in range(len(score)):
        print(i)
        # 인덱스가 k 보다 작은 경우
        if i < k:
            lowest = min(score[0:i+1])
            hallOfFame.append(lowest)
        else:
            sortedArr = sorted(score[0:i+1], reverse=True)
            lowest = sortedArr[k-1]
            hallOfFame.append(lowest)
            
    
    return hallOfFame
    